### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by shafran-andriy**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+        self.age = age
+        self.__class__.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_list = []
+    for i in range(len(people)):
+        new_list.append(Person(people[i]["name"], people[i]["age"]))
```
Comment (EdAlekseiev) [inline app/main.py:13]: Use list comprehension here

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,21 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str = "", age: int = 0) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
```
Comment (EdAlekseiev) [inline app/main.py:7]:
```suggestion
        self.people[name] = self
```

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    new_list = []
+    for i in range(len(people)):
+        new_list.append(Person(people[i]["name"], people[i]["age"]))
+
+    for obj in range(len(new_list)):
+        if "wife" in people[obj] and people[obj].get("wife") is not None:
```
Comment (EdAlekseiev) [inline app/main.py:16]:
```suggestion
        if people[obj].get("wife"):
```

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,21 @@
...
+            new_list[obj].wife = Person.people[people[obj].get("wife")]
+
+        if "husband" in people[obj] and people[obj].get("husband") is not None:
```
Comment (EdAlekseiev) [inline app/main.py:19]:
```suggestion
        if people[obj].get("husband"):
```

**Review by EdAlekseiev — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 1 above)
Comment (shafran-andriy) [reply app/main.py]: fixed

File: app/main.py (hunk 2 above)
Comment (shafran-andriy) [reply app/main.py]: fixed

File: app/main.py (hunk 3 above)
Comment (shafran-andriy) [reply app/main.py]: changed

File: app/main.py (hunk 4 above)
Comment (shafran-andriy) [reply app/main.py]: changed

**Review by EdAlekseiev — APPROVED** [review]
Good job

**Pull request closed without merging**